src/emulator/auth/state.ts

```diff
@@ -550,6 +550,45 @@ export class ProjectState {
     }
   }
 }
+
+export class AgentProjectState extends ProjectState {
+  private tenantForTenantId: Map<string, Tenant> = new Map();
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.get is added
+  getTenant(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.list is added
+  listTenants(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.create is added
+  createTenant(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.patch is added
+  updateTenant(): void {
+    return;
+  }
+
+  // TODO(lisajian): Fill in when v2.projects.tenants.delete is added
+  deleteTenant(): void {
+    return;
+  }
+}
+
+export class TenantProjectState extends ProjectState {
+  readonly tenantId: string;
+
+  constructor(projectId: string, tenantId: string) {
```

Comment:
This can be simplified to:

```ts
export class TenantProjectState extends ProjectState {
  constructor(projectId: string, readonly tenantId: string) {
    super(projectId);
  }
}
```

(This makes use of the parameter properties in typescript constructors: https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties)